packages/components/src/internal/AssayDefinitionModel.ts

```diff
@@ -297,4 +297,11 @@ export class AssayDefinitionModel extends ImmutableRecord({
 
         return columns.asImmutable();
     }
+
+    domainFileColumnNames(type: AssayDomainTypes): string[] {
```

Comment:
nit: Perhaps this should be `getDomainFileColumns` and then let the caller map out the names? I guess I'm thinking this should be similar to `getDomainColumns(type: AssayDomainTypes)`. Something like this:

```tsx
getDomainFileColumns(type: AssayDomainTypes): QueryColumn[] {
    return this.getDomainColumns(type)
        .filter(col => col.isFileInput)
        .toArray();
}
```

Then it wouldn't need to be defensive (`getDomainColumns` already is) and we leave it up to the caller if they need more column information.